src/main/java/org/jskele/libs/dao/impl/mappers/LibDaoConversionService.java

```diff
@@ -0,0 +1,59 @@
+package org.jskele.libs.dao.impl.mappers;
+
+import org.jskele.libs.dao.JsonValue;
+import org.postgresql.util.PGobject;
+import org.springframework.core.convert.ConverterNotFoundException;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.support.DefaultConversionService;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+public class LibDaoConversionService extends DefaultConversionService {
+    private final JsonValueConverter jsonValueConverter = new JsonValueConverter();
+
+    @Override
+    public boolean canConvert(TypeDescriptor sourceType, TypeDescriptor targetType) {
+        if (super.canConvert(sourceType, targetType)) {
+            return true;
+        }
+        return canConvertPGobjectToJsonValue(null, sourceType, targetType);
+    }
+
+    @Override
+    public Object convert(Object source, TypeDescriptor sourceType, TypeDescriptor targetType) {
+        try {
+            return super.convert(source, sourceType, targetType);
+        } catch (ConverterNotFoundException e) {
+            if (canConvertPGobjectToJsonValue(source, sourceType, targetType)) {
```

Comment:
i though about it, but decided to go with this approach that would allow adding custom converter by the user of this library for specific subtypes (in which case `super.convert` would succeed)